open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/VRPSubmissionPayloadValidator.java

```diff
@@ -0,0 +1,509 @@
+                        ErrorConstants.REMITTANCE_INFO_NOT_FOUND);
+            } else if (initiationOfSubmission.containsKey(ConsentExtensionConstants.REMITTANCE_INFO)
+                    && initiationParameterOfConsentInitiation.
+                    containsKey(ConsentExtensionConstants.REMITTANCE_INFO)) {
+
+                Object remittanceInformationSubmission = initiationOfSubmission
+                        .get(ConsentExtensionConstants.REMITTANCE_INFO);
+                Object remittanceInformationInitiation = initiationParameterOfConsentInitiation
+                        .get(ConsentExtensionConstants.REMITTANCE_INFO);
+
+                if (remittanceInformationSubmission instanceof JSONObject &&
+                        remittanceInformationInitiation instanceof JSONObject) {
+                    JSONObject remittanceInformationSub = (JSONObject) initiationOfSubmission
+                            .get(ConsentExtensionConstants.REMITTANCE_INFO);
+                    JSONObject remittanceInformationInit = (JSONObject) initiationParameterOfConsentInitiation
+                            .get(ConsentExtensionConstants.REMITTANCE_INFO);
+
+                    validationResult = VRPSubmissionPayloadValidator.validateRemittanceInfo
+                            (remittanceInformationSub, remittanceInformationInit);
+                    if (!Boolean.parseBoolean(validationResult.
+                            getAsString(ConsentExtensionConstants.IS_VALID_PAYLOAD))) {
+                        return validationResult;
+                    }
+                } else {
+                    return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_INVALID,
+                            ErrorConstants.INITIATION_REMITTANCE_INFO_NOT_JSON_ERROR);
+                }
+
+            } else {
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                        ErrorConstants.INITIATION_REMITTANCE_INFO_PARAMETER_NOT_FOUND);
+            }
+        }
+
+        return validationResult;
+    }
+
+    /**
+     * Validates the instruction between submission and initiation JSONObjects.
+     *
+     * @param submission The instruction submission JSONObject from submission request.
+     * @param initiation The instruction initiation JSONObject from initiation request.
+     * @return A JSONObject indicating the validation result. It contains a boolean value under the key
+     * ConsentExtensionConstants.IS_VALID_PAYLOAD, indicating whether the payload is valid. If the
+     * validation fails, it returns a JSONObject containing error details with keys defined in ErrorConstants.
+     */
+    public static JSONObject validateInstruction(JSONObject submission,
+                                                 JSONObject initiation) {
+
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
```

Comment:
Shouldn't we set this to false by default? Can you check below scenarios as well?

- submission is null, initiation is not null
- submission is not null, initiation is null
- both null

It seems for above cases we are returning `true` as validation result.

Reply:
Fixed in [8362873]

Reply:
https://github.com/wso2/financial-open-banking/pull/58#discussion_r1632618343